### PR TITLE
[DNS] Resolve API view comments

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 1]
```diff
@@ -220,38 +216,38 @@ public partial class DnsZoneResource : Azure.ResourceManager.ArmResource
...
         public virtual Azure.ResourceManager.Dns.CnameRecordCollection GetCnameRecords() { throw null; }
-        public virtual Azure.Response<Azure.ResourceManager.Dns.MXRecordResource> GetMXRecord(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.MXRecordResource>> GetMXRecordAsync(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.Dns.MXRecordResource> GetMXRecord(string MXRecordName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.MXRecordResource>> GetMXRecordAsync(string MXRecordName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:235]: Should we use `mxRecordName` here as the parameter name?

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 2]
```diff
@@ -220,38 +216,38 @@ public partial class DnsZoneResource : Azure.ResourceManager.ArmResource
...
         public virtual Azure.ResourceManager.Dns.MXRecordCollection GetMXRecords() { throw null; }
-        public virtual Azure.Response<Azure.ResourceManager.Dns.NSRecordResource> GetNSRecord(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.NSRecordResource>> GetNSRecordAsync(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.Dns.NSRecordResource> GetNSRecord(string NSRecordName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.NSRecordResource>> GetNSRecordAsync(string NSRecordName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:238]: Should we use `nsRecordName` here as the parameter name?

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 3]
```diff
@@ -263,36 +259,30 @@ public partial class DnsZoneResource : Azure.ResourceManager.ArmResource
...
-        public virtual Azure.Response<Azure.ResourceManager.Dns.MXRecordResource> Get(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Dns.MXRecordResource> CreateOrUpdate(Azure.WaitUntil waitUntil, string MXRecordName, Azure.ResourceManager.Dns.MXRecordData data, Azure.ETag? ifMatch = default(Azure.ETag?), string ifNoneMatch = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Dns.MXRecordResource>> CreateOrUpdateAsync(Azure.WaitUntil waitUntil, string MXRecordName, Azure.ResourceManager.Dns.MXRecordData data, Azure.ETag? ifMatch = default(Azure.ETag?), string ifNoneMatch = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:263]: Same as above.

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 4]
```diff
@@ -263,36 +259,30 @@ public partial class DnsZoneResource : Azure.ResourceManager.ArmResource
...
+        public virtual Azure.Response<bool> Exists(string MXRecordName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<bool>> ExistsAsync(string MXRecordName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:265]: Same as above.

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 5]
```diff
@@ -263,36 +259,30 @@ public partial class DnsZoneResource : Azure.ResourceManager.ArmResource
...
         public virtual Azure.Pageable<Azure.ResourceManager.Dns.MXRecordResource> GetAll(int? top = default(int?), string recordsetnamesuffix = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.AsyncPageable<Azure.ResourceManager.Dns.MXRecordResource> GetAllAsync(int? top = default(int?), string recordsetnamesuffix = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.MXRecordResource>> GetAsync(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.MXRecordResource>> GetAsync(string MXRecordName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:269]: Same as above

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 6]
```diff
@@ -263,36 +259,30 @@ public partial class DnsZoneResource : Azure.ResourceManager.ArmResource
...
-        public long? TtlInSeconds { get { throw null; } set { } }
     }
     public partial class MXRecordResource : Azure.ResourceManager.ArmResource
     {
         public static readonly Azure.Core.ResourceType ResourceType;
         protected MXRecordResource() { }
         public virtual Azure.ResourceManager.Dns.MXRecordData Data { get { throw null; } }
         public virtual bool HasData { get { throw null; } }
-        public static Azure.Core.ResourceIdentifier CreateResourceIdentifier(string subscriptionId, string resourceGroupName, string zoneName, string relativeRecordSetName) { throw null; }
+        public static Azure.Core.ResourceIdentifier CreateResourceIdentifier(string subscriptionId, string resourceGroupName, string zoneName, string MXRecordName) { throw null; }
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:285]: Same as above.

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 7]
```diff
@@ -303,36 +293,30 @@ public partial class MXRecordResource : Azure.ResourceManager.ArmResource
...
-        public virtual Azure.Response<bool> Exists(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<bool>> ExistsAsync(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual Azure.Response<Azure.ResourceManager.Dns.NSRecordResource> Get(string relativeRecordSetName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Dns.NSRecordResource> CreateOrUpdate(Azure.WaitUntil waitUntil, string NSRecordName, Azure.ResourceManager.Dns.NSRecordData data, Azure.ETag? ifMatch = default(Azure.ETag?), string ifNoneMatch = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:296]: Please also double check the parameter name `NSRecordName`

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs [hunk 8]
```diff
@@ -43,76 +37,84 @@ public partial class AaaaRecordResource : Azure.ResourceManager.ArmResource
...
         public virtual Azure.Response<Azure.ResourceManager.Dns.ARecordResource> Get(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.ARecordResource>> GetAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.Response<Azure.ResourceManager.Dns.ARecordResource> Update(Azure.ResourceManager.Dns.ARecordData data, Azure.ETag? ifMatch = default(Azure.ETag?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Dns.ARecordResource>> UpdateAsync(Azure.ResourceManager.Dns.ARecordData data, Azure.ETag? ifMatch = default(Azure.ETag?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public partial class BaseRecordData : Azure.ResourceManager.Models.ResourceData
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:71]: Why does `BaseRecordData` inherit from `ResourceData` instead of `RecordData`?

File: sdk/dns/Azure.ResourceManager.Dns/CHANGELOG.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 ## 1.0.0 (2022-09-29)
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/CHANGELOG.md:3]: Why did we revert the change in changelog?

File: sdk/dns/Azure.ResourceManager.Dns/README.md [hunk 10]
```diff
@@ -1,16 +1,6 @@
-# Microsoft Azure DNS management client library for .NET
+# Azure DNS Management client library for .NET
```
Comment (Yao725) [inline sdk/dns/Azure.ResourceManager.Dns/README.md:1]: Same question as above.

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs (hunk 8 above)
Comment (dvbb) [reply sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:71]: `BaseRecordData` is Autorest generated, should we add `BaseRecordData` to customization?

File: sdk/dns/Azure.ResourceManager.Dns/CHANGELOG.md (hunk 9 above)
Comment (dvbb) [reply sdk/dns/Azure.ResourceManager.Dns/CHANGELOG.md]: I copy this file from an old branch, reverted it.

File: sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs (hunk 8 above)
Comment (Yao725) [reply sdk/dns/Azure.ResourceManager.Dns/api/Azure.ResourceManager.Dns.netstandard2.0.cs:71]: I prefer to make it inherit from `RecordData`. 

Nit: I think maybe we should exchange the names of `BaseRecordData` and `RecordData`.

**Review by ArthurMa1978 — APPROVED** [review]
